### PR TITLE
Fix Clawback Description account swap

### PR DESCRIPTION
**Pull request by shawnxie999**
## High Level Overview of Change

Accounts fields are mutated because of swapping values on an object reference. We now construct a new  object using spread operator.

### Type of Change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Refactor (non-breaking change that only restructures code)
- [ ] Tests (You added tests for code that already exists, or your new feature included in this PR)
- [ ] Documentation Updates
- [ ] Release

### TypeScript/Hooks Update

- [ ] Updated files to React Hooks
- [ ] Updated files to TypeScript

### CODE REVIEW

File: src/containers/shared/components/Transaction/Clawback/Description.tsx [hunk 1]
```diff
@@ -7,7 +7,10 @@ import { formatAmount } from '../../../../../rippled/lib/txSummary/formatAmount'
 export const Description = ({ data }: TransactionDescriptionProps) => {
   const issuer = data.tx.Account
   const holder = data.tx.Amount.issuer
-  const amount = data.tx.Amount
+
+  // construct a new amount object by spreading to avoid mutating
+  // the original object reference
+  const amount = { ...data.tx.Amount }
```
Comment (mvadari) [inline src/containers/shared/components/Transaction/Clawback/Description.tsx:13]: Is this an issue in the parser too?

Comment (shawnxie999) [reply src/containers/shared/components/Transaction/Clawback/Description.tsx:13]: i dont believe so. In the parser, it calls `formatAmount`, which returns a new object rather an reference to the raw data

Comment (mvadari) [reply src/containers/shared/components/Transaction/Clawback/Description.tsx:13]: Why not call `formatAmount` in this too?

Comment (shawnxie999) [reply src/containers/shared/components/Transaction/Clawback/Description.tsx:13]: that works too. will make change

**Review by ckniffen — APPROVED** [review]

**Review by mvadari — APPROVED** [review]

**Pull request merged**